### PR TITLE
updated the Modal web component in the vaos Application

### PR DESCRIPTION
**Pull request by simiadebowale**
## Description
Platform Design System team built a Modal web component. This PR replaces the existing Modal component with the updated Modal web component.

## Original issue(s)
[department-of-veterans-affairs/va.gov-team#38608](https://app.zenhub.com/workspaces/vaos-team-603fdef281af6500110a1691/issues/department-of-veterans-affairs/va.gov-team/38608)

## Testing done
updated unit tests

## Acceptance criteria
- Modal component is updated across vaos application and functioning as expected.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/cancel/CancelCOVIDVaccineModal.jsx [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
   return (
-    <Modal
+    <VaModal
       id="cancelAppt"
       status="warning"
       visible
-      onClose={onClose}
-      title="You need to call your VA medical center to cancel this appointment"
+      onCloseEvent={onClose}
+      modalTitle="You need to call your VA medical center to cancel this appointment"
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/cancel/CancelCOVIDVaccineModal.jsx:13]: Add role.

**Comment by vbahinwillit** [comment]
LGTM. Just a couple of general items. I would add the role to all of the modals and fix the 'quickfix' eslint errors. We've been fixing all eslint errors whenever we touch files since the platform turned on additional rules.

File: src/applications/vaos/appointment-list/components/cancel/CancelAppointmentFailedModal.jsx [hunk 2]
```diff
@@ -47,6 +49,6 @@ export default function CancelAppointmentFailedModal({
           </>
         )}
       </p>
-    </Modal>
+    </VaModal>
   );
 }
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/cancel/CancelAppointmentFailedModal.jsx:54]: You can add propTypes to get rid of the linting warnings just like you would find in `CancelAppointmentConfirmationModal.jsx`

**Review by jenniemc — CHANGES_REQUESTED** [review]
Hi Simi, looks pretty good. Added some suggestions to fix linting warnings and adding attributes to the VAModal component.

**Review by jenniemc — APPROVED** [review]
LGTM!

**Pull request merged**